libgc/configure.ac

```diff
@@ -104,6 +104,10 @@ case "$THREADS" in
 	AC_DEFINE(GC_AIX_THREADS)
 	AC_DEFINE(_REENTRANT)
 	;;
+     *-*-os400*)
+	AC_DEFINE(GC_AIX_THREADS)
+	AC_DEFINE(_REENTRANT)
+	;;
      *-*-haiku*)
```

Comment:
undo this change in `libgc`, we'll likely have a submodule for boehm GC soon, so this would be lost anyway (/cc @joncham )